lib/sda/host_msg.go

```diff
@@ -56,6 +63,8 @@ func (t *Token) Id() uuid.UUID {
 
 // Return a new Token contianing a reference to the given TreeNode
 func (t *Token) ChangeTreeNodeID(newid uuid.UUID) *Token {
+	tokenMutex.Lock()
+	defer tokenMutex.Unlock()
```

Comment:
The race detector told me so... I think the problem comes if you create an Id and then call ChangeTreeNodeID on it. But I didn't follow the exact call-path to know how this happens.